### ISSUE TITLE
Comments on Message receive and wait channel misleading

### ISSUE THREAD

**Issue by WhisperingChaos** [gobridge/concurrency-patterns#4]
Certain comments in the  [sequencing pattern](https://github.com/gobridge/concurrency-patterns/blob/c04c593639aa528fa33c58bb0e5213c5486bed7d/4-multiplexing/sequencing/sequencing.go) imply Joe's goroutine will generate the first received Message struct followed by Ann's:

https://github.com/gobridge/concurrency-patterns/blob/c04c593639aa528fa33c58bb0e5213c5486bed7d/4-multiplexing/sequencing/sequencing.go#L35-L42

However, the code doesn't explicitly define a [Happens Before](https://golang.org/ref/mem#tmp_2) dependency to ensure this ordering.  Therefore, it's possible that Ann's Message struct will precede Joe's resulting in the variable _msg1_ containing Ann's message, not Joe's as suggested by the comments. 

If a strict Round-robin ordering is desired, the following [implementation](https://play.golang.org/p/VbjVK-z50Ws) delivers the same outcome while eliminating the wait abstraction and its associated code.

**Comment by ardan-bkennedy** [comment]
Yes, this is concurrent code (out of order execution) so the goroutine started for Ann could execute before Joe. However, the code you listed in Main is running on it's on goroutine and is sequential. What we see is the order of execution for that goroutine and if Ann executes first underneath, it won't change the output of main.

**Comment by WhisperingChaos** [comment]
> Yes, this is concurrent code (out of order execution) so the goroutine started for Ann could execute before Joe.

Agreed.

>However, the code you listed in Main is running on it's on goroutine and is sequential.

Agreed.

Given the _fanIn_ function code: 

https://github.com/gobridge/concurrency-patterns/blob/c04c593639aa528fa33c58bb0e5213c5486bed7d/4-multiplexing/sequencing/sequencing.go#L51-L61

- Does not directly cause these goroutines to execute,
- and there isn't any code within them nor code external to them that orders/coordinates their **initial execution**. 

Therefore, even though _main_ is sequentially executing its loop, it's left to the golang runtime scheduler to determine the execution order of the (concurrent) goroutines that that feed _Message_(s) to the common channel referenced by _main_.  Since the runtime scheduler doesn't, by itself, guarantee the execution ordering of goroutines, the output of the sequential loop can change.  That is, the first _Message_ from Ann can appear before Joe's first one.

Finally, the evidence of this behavior in a running example.  A trivial change that reduces the scope of the variable _waitForIt_ in the function _boring_ shouldn't affect the ordering of the output messages.  However, it does.  Please run the following [playground code](https://play.golang.org/p/KTEh3TRPW4g) a few times to witness the "Ann 0" message output before "Joe 0".
